x-pack/plugins/infra/server/lib/infra_types.ts

```diff
@@ -5,20 +5,20 @@
  * 2.0.
  */
 
-import { Logger } from '@kbn/logging';
+import type { Logger } from '@kbn/logging';
 import type { IBasePath } from '@kbn/core/server';
-import { handleEsError } from '@kbn/es-ui-shared-plugin/server';
+import type { handleEsError } from '@kbn/es-ui-shared-plugin/server';
 import type { AlertsLocatorParams } from '@kbn/observability-plugin/common';
-import { ObservabilityConfig } from '@kbn/observability-plugin/server';
-import { LocatorPublic } from '@kbn/share-plugin/common';
-import { RulesServiceSetup } from '../services/rules';
-import { InfraConfig, InfraPluginStartServicesAccessor } from '../types';
-import { KibanaFramework } from './adapters/framework/kibana_framework_adapter';
-import { InfraFieldsDomain } from './domains/fields_domain';
-import { InfraLogEntriesDomain } from './domains/log_entries_domain';
-import { InfraMetricsDomain } from './domains/metrics_domain';
-import { InfraSources } from './sources';
-import { InfraSourceStatus } from './source_status';
+import type { ObservabilityConfig } from '@kbn/observability-plugin/server';
+import type { LocatorPublic } from '@kbn/share-plugin/common';
+import type { RulesServiceSetup } from '../services/rules';
+import type { InfraConfig, InfraPluginStartServicesAccessor } from '../types';
+import type { KibanaFramework } from './adapters/framework/kibana_framework_adapter';
+import type { InfraFieldsDomain } from './domains/fields_domain';
+import type { InfraLogEntriesDomain } from './domains/log_entries_domain';
+import type { InfraMetricsDomain } from './domains/metrics_domain';
+import type { InfraSources } from './sources';
+import type { InfraSourceStatus } from './source_status';
```

Comment:
**nit:** I didn't verify each one of these imports and the others I've seen switched to the `import type` signature, but I assume not all of them have runtime types.

If that's the case, most of these imports don't require importing using `import type` because they are purely type declarations and don't have any associated runtime values. We can simply use a regular import statement to bring them into the code.

The `import type` syntax is good for excluding runtime code we have associated with classes, enums etc.. when we only need to use the types and we want so to exclude the code from the bundle.